src/main.rs

```diff
@@ -114,62 +108,92 @@ struct Opt {
     dry_run: bool,
 }
 
-#[derive(Clone, Serialize, Deserialize, Debug)]
+#[derive(Clone, Debug)]
 struct Inventory {
-    inv: HashMap<CacheKey, CacheValue>,
+    inv: HashMap<CommandInputHashes, LastUsedAndSize>,
     cache_path: PathBuf,
+    regex: Regex,
 }
 
 impl Inventory {
+    fn load_entry(&mut self, path: PathBuf) -> Result<()> {
+        let string_path = path.to_string_lossy().to_string();
+        let caps = self
+            .regex
+            .captures(&string_path)
+            .ok_or(eyre!(string_path.clone()))?;
+
+        let command_hash = u64::from_str_radix(&caps[1], 16)?;
+        let input_hash = u64::from_str_radix(&caps[2], 16)?;
+
+        let metadata = path.metadata()?;
+
+        self.inv.insert(
+            CommandInputHashes {
+                command_hash,
+                input_hash,
+            },
+            LastUsedAndSize {
+                last_used: metadata.accessed()?,
```

Comment:
I think you need `max(metadata.created(), metadata.modified())`, as hashing the file (reading it) advances the access time, and then we'd never use the cache